### PR TITLE
Redirection to shared url after login

### PR DESCRIPTION
**Pull request by ankursinghal2005**
Use Case:

1. User is not logged in
2. User tries to access shared(bookmark) link
3. User is redirected to the login form
4. After entering his credentials, user is redirected to shared(bookmark) link

This PR fixes below issue-:
https://github.com/apache/incubator-superset/issues/5243

### CODE REVIEW

**Comment by davidhassan** [comment]
Thanks for this PR.

**Comment by arpit-agarwal** [comment]
@williaster @mistercrunch  can you have a look at this PR.

This changes looks good to me and working as intended

File: superset/superset_decorators.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+# Get target_url from session object and redirect to that url
+def redirect_to_target_url(f):
+    def wraps(self, *args, **kwargs):
+        if 'target_url' in session:
+            if session['target_url'] != '':
+                url = session['target_url']
+                session.pop('target_url')
+                return redirect(url)
```
Comment (mistercrunch) [inline superset/superset_decorators.py]: I don't think this is secure:
https://www.owasp.org/index.php/Top_10_2013-A10-Unvalidated_Redirects_and_Forwards
http://flask.pocoo.org/snippets/62/

**Review by mistercrunch — CHANGES_REQUESTED** [review]

**Comment by mistercrunch** [comment]
@betodealmeida I thought you had worked on something like this in the past (?)

**Comment by betodealmeida** [comment]
@mistercrunch I fixed this upstream in FAB: https://github.com/dpgaspar/Flask-AppBuilder/pull/910

File: superset/superset_decorators.py (hunk 1 above)
Comment (betodealmeida) [reply superset/superset_decorators.py]: Right. If you look at https://github.com/dpgaspar/Flask-AppBuilder/pull/910 (and how we do it in Airflow) you should cryptographically sign the target URL to make sure that a compromised auth provider doesn't send you to a malicious host.

**Review by betodealmeida — CHANGES_REQUESTED** [review]
As @mistercrunch mentioned, this approach is not safe. I've fixed this upstream, I'll ping @dpgaspar to review it and merge it.

File: superset/superset_decorators.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
[the remark this answers is not in the document]
Comment (ankursinghal2005) [reply superset/superset_decorators.py]: sure

File: superset/superset_decorators.py (hunk 1 above)
Comment (ankursinghal2005) [reply superset/superset_decorators.py]: Thanks @betodealmeida @mistercrunch for suggestion, I will check and update..

**Comment by ankursinghal2005** [comment]
@betodealmeida @mistercrunch Please find my comments here.

https://github.com/dpgaspar/Flask-AppBuilder/pull/910/files#r262546216

**Comment by mistercrunch** [comment]
Please reopen when the security concerns are addressed

**Comment by agabardo** [comment]
Why do people close unsolved issues? Has this problem being fixed?

**Pull request closed without merging**